concert/ext/tangoservers/walker.py

```diff
@@ -182,40 +183,54 @@ async def write_sequence(self, name):
             self.set_state(DevState.FAULT)
             raise
 
-
-
-
     async def consume(self, walker: DirectoryWalker) -> None:
+        """Defines an internal consumer coroutine to write incoming stream"""
         with PerformanceTracker() as pt:
             total_bytes = await walker.write(self._receiver.subscribe())
             pt.size = total_bytes * q.B
 
     @DebugIt()
     @command(
-        dtype_in=str,
-        doc_in="file name for the log file"
+        dtype_in=(str,),
+        dtype_out=str,
+        doc_in="log_name, log_level and file_name for logging",
+        doc_out="absolute log file path as an identifier for the logger"
     )
-    async def open_log_handler(self, file_name: str) -> None:
-        self._log_handler = logging.FileHandler(os.path.join(self._current, file_name))
-        self._logger.addHandler(self._log_handler)
+    async def register_logger(self, args: Tuple[str, str, str]) -> str:
+        logger_name, log_level, file_name = args[0], int(args[1]), args[2]
+        # NOTE: log_path functions as unique identifier for a logger instance as well as the
+        # absolute path for the log file, where logs should be written.
+        log_path: str = os.path.join(self._current, file_name)
+        handler = logging.FileHandler(log_path)
+        logger = logging.Logger(logger_name, log_level)
+        logger.addHandler(handler)
+        self._log_handlers[log_path] = handler
```

Comment:
Could you raise an exception `if log_path in self._log_handlers`? It will prevent some unexpected behavior. It's also one more tiny test.